File: src/org/foonugget/kdict/data/AppDatabaseOpenHelper.java

package org.foonugget.kdict.data;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.google.inject.Inject;
import com.google.inject.Singleton;

@Singleton
public class AppDatabaseOpenHelper extends SQLiteOpenHelper {

    private static final int VERSION = 1;

    private static final String NAME = "cache.db";

    @Inject
    public AppDatabaseOpenHelper(final Context context) {
        super(context, NAME, null, VERSION);
    }

    @Override
    public void onCreate(final SQLiteDatabase db) {
        db.execSQL("CREATE TABLE search_history (id INTEGER PRIMARY KEY, search_string TEXT)");
    }

    @Override
    public void onUpgrade(final SQLiteDatabase db, final int oldVersion, final int newVersion) {
        /* no upgrades yet */
    }

}

File: app/src/main/java/org/foonugget/kdict/data/AppDatabaseOpenHelper.kt
package org.foonugget.kdict.data

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class AppDatabaseOpenHelper(context: Context) : SQLiteOpenHelper(context, NAME, null, VERSION) {

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL("CREATE TABLE search_history (id INTEGER PRIMARY KEY, search_string TEXT UNIQUE)")
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        db.execSQL("DROP TABLE IF EXISTS search_history")
        db.execSQL("CREATE TABLE search_history (id INTEGER PRIMARY KEY, search_string TEXT UNIQUE)")
    }

    companion object {
        private const val VERSION = 3
        private const val NAME = "cache.db"
    }
}
